### DISCUSSION TITLE
Is there support for various levels of comprehension?

### DISCUSSION THREAD

**Discussion by crnormand** [mclemente/fvtt-module-polyglot#232] [Q&A] [answered]
The GURPS game system has "advantages" for both Written and Spoken versions of a language, and at various levels (Literate / Native, Accented, Semi-Literate, Broken).

Does Polygot offer different levels of "translation" based on a "comprehension" level?   Or is it a boolean (translated or untranslated)?

Also, is there a way to indicate spoken vs written?

**Comment by mclemente** [comment, marked as answer]
>Does Polygot offer different levels of "translation" based on a "comprehension" level? Or is it a boolean (translated or untranslated)?

It's entirely binary. The scrambling just makes a random string with different fonts, so showing different levels of understanding would probably generate unreadable text (think reading a string with two different alphabets mixed together).

>Also, is there a way to indicate spoken vs written?

It's doable, it's up to you. It can be something like `Language (Common, Spoken)` or any variation. Anything that can be caught on a Regex. it just needs a pattern.

**Comment by crnormand** [comment]
After viewing the code, I see that there is known_languages and literate_languages.    Thank you for your answer, and I should have a PR ready in the new few days.
